fuel/datasets/svhn.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+import os
+
+from fuel import config
+from fuel.datasets import H5PYDataset
+from fuel.transformers.defaults import uint8_pixels_to_floatX
+
+
+class SVHN(H5PYDataset):
+    """The Street View House Numbers (SVHN) dataset.
+
+    SVHN [SVHN] is a real-world image dataset for developing machine
+    learning and object recognition algorithms with minimal requirement
+    on data preprocessing and formatting. It can be seen as similar in
+    flavor to MNIST [LBBH] (e.g., the images are of small cropped
+    digits), but incorporates an order of magnitude more labeled data
+    (over 600,000 digit images) and comes from a significantly harder,
+    unsolved, real world problem (recognizing digits and numbers in
+    natural scene images). SVHN is obtained from house numbers in
+    Google Street View images.
+
+    .. [SVHN] Yuval Netzer, Tao Wang, Adam Coates, Alessandro Bissacco,
+       Bo Wu, Andrew Y. Ng. *Reading Digits in Natural Images with
+       Unsupervised Feature Learning*, NIPS Workshop on Deep Learning
+       and Unsupervised Feature Learning, 2011.
+
+    .. [LBBH] Yann LeCun, Léon Bottou, Yoshua Bengio, and Patrick Haffner,
+       *Gradient-based learning applied to document recognition*,
+       Proceedings of the IEEE, November 1998, 86(11):2278-2324.
+
+    Parameters
+    ----------
+    which_format : 1 or 2
+        SVHN format 1 contains the full numbers, whereas SVHN format 2
+        contains cropped digits. **Note: at the moment, only format 2
+        is available through Fuel.**
+    which_set : 'train', 'test' or 'extra'
```

Comment:
Same here, should be `{'traing', 'test', 'extra'}` according to NumPy docstring standards I think